flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaEvolutionUtil.java

```diff
@@ -304,4 +305,52 @@ private static int indexOf(DataField dataField, LinkedHashMap<Integer, DataField
         throw new IllegalArgumentException(
                 String.format("Can't find data field %s", dataField.name()));
     }
+
+    /**
+     * Create converter mapping from table fields to underlying data fields. For example, the table
+     * and data fields are as follows
+     *
+     * <ul>
+     *   <li>table fields: 1->c INT, 6->b STRING, 3->a BIGINT
+     *   <li>data fields: 1->a BIGINT, 3->c DOUBLE
+     * </ul>
+     *
+     * <p>We can get the column types (1->a BIGINT), (3->c DOUBLE) from data fields for (1->c INT)
+     * and (3->a BIGINT) in table fields through index mapping [0, -1, 1], then compare the data
+     * type and create converter mapping.
+     *
+     * <p>/// TODO should support nest index mapping when nest schema evolution is supported.
+     *
+     * @param tableFields the fields of table
+     * @param dataFields the fields of underlying data
+     * @param indexMapping the index mapping from table fields to data fields
+     * @return the index mapping
+     */
+    public static DataValueConverter[] createConvertMapping(
```

Comment:
Maybe we can make converters consistent with indexMapping?
- if (dataField.type().equals(tableField.type())), we can return an identity converter.
- If all converters are identity, we can make this returning null array.

In this way, we can remove null check for the element of this array.